actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -438,7 +438,7 @@ def form_for(record, options = {}, &block)
           model       = nil
           object_name = record
         else
-          model       = convert_to_model(record)
+          model       = record
```

Comment:
I was hoping to avoid mixing too many refactoring changes with changes to the behavior. Inlining `apply_form_for_options!` feels reasonable, but might even be better suited as a follow-up PR.

While the parallel declarations might be confusing, it's a pattern that pre-dates this change so I felt like it was worth sticking with.

If it's decided that this _isn't_ a breaking change, and that it's a worthwhile bugfix, I'd feel more comfortable broadening the scope to include some variable clean up. What do you think?